config/charts/inferencepool/README.md

```diff
@@ -43,6 +45,50 @@ inferenceExtension:
     FEATURE_FLAG_ENABLED: "true"
 ```
 
+#### Advanced Environment Variables
+
+For more complex environment variable configurations, including secrets, configmaps, and field references, use the `envVars` and `envFrom` fields:
+
+```yaml
+# values.yaml
+inferenceExtension:
+  # Simple key-value environment variables
+  env:
```

Comment:
Why not have env accept any yaml? then it's up to the user to use it properly.

E.g., in the app-template, env can take:
```
{{- toYaml . | nindent X }} # X should probably be 8
        {{- end }}
```